Add how to make a release notes (#3793)
SpikeInterface/spikeinterface: doc/development/development.rst

to describe your sorter.

Finally, make a pull request so we can review the code and incorporate into the sorters module of SpikeInterface!

How to make a release
---------------------

Checklist
^^^^^^^^^

> * Pyproject: check that the version is ahead of currently release. Also, comment out the @ (git dependencies)

Reviewer: ```suggestion
* pyproject.toml: check that the version is ahead of currently release. Also, comment out the @ (git dependencies)
```
by the time someone is making a release they would probably know, but just to be extra clear

Author: Yes!

> * pyproject.toml: check that the version is ahead of currently release. Also, comment out the @ (git dependencies)

Reviewer 2: ```suggestion
* pyproject.toml: check that the version is ahead of current release. Also, comment out the @ (git dependencies)
```

> * In the top level ``__init__`` uncomment ``DEV_MODE`` (this is used for the docker installations)

Reviewer: Do we want to say the location rather than just say top level?

Author: It would be better yes.

Author: Done.

* Create a new release note for the appropriate version on doc/releases/new_version_tag.

> There can be large releases like:

Reviewer 2: ```suggestion
There can be large releases like:

```

To be consistent with the whitespace below 'Which contain a section called "Main Changes" and minor releases which include only bug fixes like:

'

Author: Thanks.

``doc/releases/0.101.0.rst``

Which contain a section called "Main Changes" and minor releases which include only bug fixes like:

``doc/releases/0.101.2.rst``

To collect all the PRs and bug fixes we have a script in:
``doc/scripts/``
called ``auto-release-notes.sh``. Run it with ``bash auto-release-notes.sh`` and it will create the release notes for the module specific changes.

> To run the script you will need to authorize to GitHub for the first time.

Reviewer: What does this mean? (ie I don't think this is quite clear if this is someone new to the team :) )

> To run the script you will need to authorize to GitHub for the first time.

Reviewer 2: Maybe something like
```suggestion
The first time you run the script, GitHub will guide you through an authorization process if you've not already done so.
```
to make it clear that you don't need to do anything special: Github will guide you through the authorization process?

Reviewer: I think this is a clearer rewriting!

Author: Way better, thanks!

The signature of the script is:

.. code-block:: bash

>     bash auto-release-notes.sh <start_date> <end_date>

Reviewer 2: Can you give an example, and use an example where the order of day and month are obvious - I can never remember how to write dates...

Reviewer: Actually this is super key. Great point! US style vs European style!

Author: Done.

Where the start date is the date of the last release and the end date is the current date.

> The date of the last release can be found from PyPI.

Reviewer: ```suggestion
The date of the last release can be found on PyPI.
```
This is more just a gut feeling for the preposition. You typically say something is on a website unless you're emphasizing that you're scraping from the website. So from isn't wrong, but on just feels more right in this case :)

Author: Yes, thanks!

to describe your sorter.

Finally, make a pull request so we can review the code and incorporate into the sorters module of SpikeInterface!

How to make a release
---------------------

Checklist
^^^^^^^^^
* pyproject.toml: check that the version is ahead of current release. Also, comment out the @ (git dependencies)

> * In the top level ``__init__`` (located on ``src/spikeinterface/__init__.py``) uncomment ``DEV_MODE`` (this is used for the docker installations)

Reviewer: ```suggestion
* In the top level ``__init__`` (located on ``src/spikeinterface/__init__.py``) uncomment ``DEV_MODE`` (this is used for the docker installations)
```

The `dev_mode` is completely clear because we have the two lines one with True and one with False. To be clear we should say to uncomment the `DEV_MODE=False` and to comment out the `DEV_MODE=True`

Reviewer: isn't * sorry my typo.

* Create a new release note for the appropriate version on doc/releases/new_version_tag.

There can be large releases like:

``doc/releases/0.101.0.rst``

Which contain a section called "Main Changes" and minor releases which include only bug fixes like:

``doc/releases/0.101.2.rst``

To collect all the PRs and bug fixes we have a script in:
``doc/scripts/``
called ``auto-release-notes.sh``. Run it with ``bash auto-release-notes.sh`` and it will create the release notes for the module specific changes.

The first time you run the script, GitHub will guide you through an authorization process if you've not already done so.

The signature of the script is:

.. code-block:: bash

    bash auto-release-notes.sh <start_date> <end_date>

Where the start date is the date of the last release and the end date is the current date.

The date of the last release can be found on `PyPI <https://pypi.org/project/spikeinterface/>`_.

> As an specific example:

Reviewer: Or one could say:
```
As an example:
```
```
For example:
```

to describe your sorter.

Finally, make a pull request so we can review the code and incorporate into the sorters module of SpikeInterface!

How to make a release
---------------------

Checklist
^^^^^^^^^
* pyproject.toml: check that the version is ahead of current release. Also, comment out the @ (git dependencies)

> * In the top level ``__init__`` (located at ``src/spikeinterface/__init__.py``) uncomment ``DEV_MODE`` (this is used for the docker installations)

Reviewer 3: ```suggestion
* In the top level ``__init__`` (located at ``src/spikeinterface/__init__.py``) set ``DEV_MODE`` to ``False`` (this is used for the docker installations)
```

* Create a new release note for the appropriate version on doc/releases/new_version_tag.

There can be large releases like:

``doc/releases/0.101.0.rst``

Which contain a section called "Main Changes" and minor releases which include only bug fixes like:

``doc/releases/0.101.2.rst``

To collect all the PRs and bug fixes we have a script in:
``doc/scripts/``
called ``auto-release-notes.sh``. Run it with ``bash auto-release-notes.sh`` and it will create the release notes for the module specific changes.

The first time you run the script, GitHub will guide you through an authorization process if you've not already done so.

The signature of the script is:

.. code-block:: bash

    bash auto-release-notes.sh <start_date> <end_date>

> Where the start date is the date of the last release and the end date is the current date.

Reviewer 3: maybe add that <date> is formatted as YYYY-MM-DD?

> Where the start date is the date of the last release and the end date is the current date.

Author: ```suggestion
Where the start date is the date of the last release and the end date is the current date. Dates are in YYYY-MM-DD format
```

Author: @alejoe91

The date of the last release can be found on `PyPI <https://pypi.org/project/spikeinterface/>`_.

As a specific example:
.. code-block:: bash

    bash auto-release-notes.sh 2025-02-19 2025-03-24

* Finish the release notes and merge
* Locally tag the main branch with the newly merged release notes with the new version
* Push the tag to the remote repository which will trigger the release action (.github/workflows/publish-to-pypi.yml)

> * Do an after-release PR:

Author: ```suggestion
* Do an after-release `PR <https://github.com/SpikeInterface/spikeinterface/pull/3828/files>`_:
```

    - Uncomment the git installs in pyproject

>     - Modify `DEV_MODE` in the top level ``__init__`` (located at ``src/spikeinterface/__init__.py``)

Reviewer 3: ```suggestion
    - Modify ``DEV_MODE`` to ``True`` in the top level ``__init__`` (located at ``src/spikeinterface/__init__.py``)
```
